### DISCUSSION TITLE
Introduce yourself!

### DISCUSSION THREAD

**Discussion by maxammann** [maplibre/maplibre-rs#49] [General]
You want to engage in the development of maplibre-rs? You are unsure where and how to start?
Introduce yourself here and I'm sure we can help you to get started. Non-developers as well as developers wanted alike.

**Comment by maxammann** [comment]
So let me start: I’m Max and I’m a software engineer and aspiring security researcher. I enjoy writing open-source software. I regularly [contribute](https://maxammann.org/projects/) to various open-source projects. I’m also a passionate photographer.

I started this project out of curiosity for vector maps and Rust. I'm looking forward to test the limits of the current Rust geo ecosystem!

**Comment by nyurik** [comment]
I'm Yuri, have been in the open source for quiet a few years now, first focusing on Wikipedia (I created Wikipedia API, maps, graphs, data pages, contributed to Wikidata, ...).  I also co-found MapLibre, and now spend most of my time on cartography, MapLibre, OpenMapTiles, Planetiler, and other map-related technologies.

Looking forward for MapLibre-rs to take over the world of mapping :)

**Comment by FabianWildgrube** [comment]
I'm Fabian, a recent software engineering graduate and computer graphics enthusiast (though not a pro at all!). In my day-to-day job I do R&D at a major hardware vendor and work mostly with C++. I'm still learning Rust and this is my first time contributing (more than translations and typos) to an open-source project and I'm very excited to get into it 😄 

Maps are everywhere and having a truly cross-platform solution that is not proprietary is something very much needed imho. I'm excited to see where MapLibre-rs is headed 😄

**Comment by Drabble** [comment]
I'm Antoine, I did a master's degree in computer Science and now I work in R&D at the University of Applied Sciences of Western Switzerland. I found a new interest in Geoinformatics when @bchapuis introduced me. I have worked on two projects since then.

- [Baremaps](https://github.com/baremaps/baremaps) is a toolkit for creating custom vector tiles.
- A multi-platform map renderer for vector tiles based on Rust and Webgpu. I'll attach a screenshot below.

<img width="400" alt="Screenshot 2022-04-28 at 09 32 05" src="https://user-images.githubusercontent.com/3996615/165700807-7584c68a-cd77-4b54-bb03-044f29238bf4.png">

I am still new to Rust and contributing to open source projects. But I'm extremely eager to learn as much as possible and help with maplibre-rs.

**Comment by quillasp** [comment]
I'm Soulaymane, currently finishing my last bachelor semester in computer science at the University of Applied Sciences of Western Switzerland. I'm a Rust enthusiast but did nothing apart from school project with it. That would be my first real contribution to an open source project. 

I'm also really interested in everything related to computer graphics, even though I'm still a beginner, so I hope I will be able to be of help.

**Comment by DerKarlos** [comment]
About me? German, a bit dyslexic and hectic, Science Fiction, Raumpatrouille, Robots, R&D+testing Industrial Automatisation, OSM.
I almost lost interest in OSM and started 3D/VR coding with Three-JS. See: http://www.ac1000.de/KK/kk

As Pokémon GO popped up, I got the idea to mimic it by rendering native OSM data.
But it endet as a 3D rendere for OSM tagging. See https://wiki.openstreetmap.org/wiki/OSM_go
Next I joined OSM2World to code a web frontend. See https://twitter.com/OSM__go/status/1365622228289785861
I changed the code to Babylon-JS and to TypeScript.

When WASM emerged, I was enthusiastic about it and even used it with CPP in OSM_go.
When Rust emerged, I was enthusiastic about and transcoded parts of the OSM2World frontend to it.
  First, I used the crate three-d, next wgpu directly and rend3 (+tracy), at last also some bevy.
  See: https://github.com/DerKarlos/O2W-Rust-WASM (Be kind, it is my first Rust project :-)
Since vector rendering emerged, I am enthusiastic about it. "Everything gets possible".

The tools for OSM evolves evolutionary. The Crate concept of Rust seems a bit more cooperative.
I wondered, how much of the OSM tools exist or could be done in Rust.
And now I spotted Maplibre-TS! I would like to see it, becoming the next-gen mapping solution.

May be I could help. Do you have some simple stupide things to do?
May be do tests? I use a Macbook Air, iPad and iPhone.
Or check errors: My garden shade is rendered as a rhomb. Tile data or calculation limit?
May be I could add functions: Click on a map object should show the OSM data of it.
Later even the OSM2World code may added (as an overlay).

I am bad with tooling and may have questions: What IDE do you use, also for Debugging?

**Reply by nyurik** [reply]
maybe create a rust wrapper around maplibre-native, and figure out if we can use some of CPP code as a module for maplibre-rs (until we migrate it over)?  I don't think it is very simple though :)  Or maybe try to integrate maplibre-rs into maplibre/martin as a server-side rendering?  Also not that simple, but definitely a lot of fun and needed :)

**Reply by maxammann** [reply]
> Or maybe try to integrate maplibre-rs into maplibre/martin as a server-side rendering?

That would definitely be possible! Performance wise, the architecture of headless rendering should be redone, but essentially it already works.

**Reply by maxammann** [reply]
> What IDE do you use, also for Debugging?

CLion because of the great debugger support. VSCode just doesn't work for me that well as a whole.
Rust-analyzer in VSCode is better in some areas though.


> And now I spotted Maplibre-TS! I would like to see it, becoming the next-gen mapping solution.

haha I think you mean maplibre-rs! :D

Welcome Karlos! The work you have done previously looks very cool. Also the tech-stack roughly seems to match that of maplibre-rs. Looking forward to hack on the codebase with you!

Right now I'm using the time, in which not a lot of contributions are coming in, to refactor the project. My current goal is to add per-file and per-function documentation within the next 2 weeks.

Right now there are still some basic challenges which need to be solved.

* How is the map state stored? Should use use Rc? Should we introduce an event system to ease mutability problems?

**Comment by DerKarlos** [comment]
Sure, I ment maplibre-rs  %-)
CLion is not open source, needs money. And as it uses IntellyJ, I may just use IntellyJ (or VRCode).
Doc will be really finde. An overview, the modules, the data. I could offer an outside view :)
Is the code of -rs in parts a 1:1 recoding of -js?  Is there a usefull doc for maplibre-js/native
How can I find you all on Matrix? (Or others?)
"Sometimes I even don't understand the problem"

**Reply by maxammann** [reply]
It's a completely new project with the aim of fitting the maplibre ecosystem.

Im relatively sure we can get you a license once  ypu contributed to the project. But a lot of devs use VSCode so that should be fine.

There is a matrix chat: https://matrix.to/#/#maplibre:matrix.org

**Reply by nyurik** [reply]
I think we can ask intellij for a FOSS license for that.
P.S. please use "write a reply" box, not the "write a comment" to keep it in the therad.

**Reply by maxammann** [reply]
Naw, we are funded by commercial conpanies. So we are excluded from that program.

**Reply by nyurik** [reply]
Well, so is Wikipedia, but they still get IntellIJ licenses.

**Comment by HideBa** [comment]
Hello, I'm HideBa from Japan, and currently in charge of the development and management of https://github.com/reearth . I'm enthusiastic about maps, geospatial data, and 3D. 
I just found this project and am very interested in joining here. 😄

**Reply by maxammann** [reply]
Very cool! I'm very limited on the time I can invest in the project right now. But it will get more again!

**Reply by pka** [reply]
Hi, heard a lot about your platform at FOSS4G in Prizren! Great, that you're also interested in Rust.

**Comment by TheToddmeister** [comment]
Heyo
Im late to the party
My name is Tormod. I've been fiddling around with Maplibre for a while, but i'd like to contribute :)

**Reply by maxammann** [reply]
Welcome! Let me know if you need anything
